### PR TITLE
feat(packfile): adds goroutines to fixAndBreakChains iterator

### PR DESCRIPTION
**Pull request by apriendeau**
This PR adds goroutine support to fixing and breaking chains due to large repos hitting a stack over-limit when the chain is too long.

### CODE REVIEW

**Review by johncoder — COMMENTED** [review]
Is this case worth adding a unit test for?

**Comment by Nivl** [comment]
You're not using workers? I'm not a huge fan of just spawning go-routines in a loop, especially when the number of iterations is defined at runtime and based on things we don't control.

**Comment by apriendeau** [comment]
@Nivl having the set number of workers caused bugs in smaller repos. For example, it required 200 threads on large repo but then would crash with a Stack Overflow limit on a smaller set which then I had to change the number. 

With that scenario above, It would need to be dynamically set and having it set by the limit of objects to pack seemed to yield the best results when setting it dynamically. It would be the same as setting the workers == len(objects). If you prefer, we can set it directly outside of the for loop.

**Comment by Nivl** [comment]
> @Nivl having the set number of workers caused bugs in smaller repos. For example, it required 200 threads on large repo but then would crash with a Stack Overflow limit on a smaller set which then I had to change the number.

That's sounds... weird. What if we set a maximum of 200 workers, and reduce it to `len(objects)` if we have less than 200 objects? My main issue is that I don't want us to spawn 1 million go-routine if we have one million objects. Ultimately I want us to have a hard limit (even arbitrary), _just in case_.

**Comment by apriendeau** [comment]
What is the benefit to that as go is designed to handle that amount and is very efficient at scheduling them? 

Edit: I am looking for a real world scenario to handicap the language here and why would want to do that over letting Go control it.

**Comment by Nivl** [comment]
goroutines have a memory and CPU cost, and since this command will eventually be run in abstractd I want us to be careful with this kind of things. Spawning a ton of goroutines can actually lead to poorer performance, and if we can avoid consuming Gbs of data just in goroutines, that would probably be a good things since this commands already uses a lot.

**Review by johncoder — APPROVED** [review]
Let's call this good. I think it'll get us past our current issues, but I anticipate having to come back to this at some point.

File: plumbing/format/packfile/delta_selector.go [hunk 1]
```diff
@@ -138,16 +139,33 @@ func (dw *deltaSelector) fixAndBreakChains(objectsToPack []*ObjectToPack) error
...
 }
 
+func (dw *deltaSelector) fixAndBreakChainsWorker(id int, objectsToPack map[plumbing.Hash]*ObjectToPack, jobs <-chan *ObjectToPack, results chan<- error) {
+	for j := range jobs {
+		err := dw.fixAndBreakChainsOne(objectsToPack, j)
+		if err != nil {
+			results <- err
+		} else {
+			results <- nil
+		}
```
Comment (Nivl) [inline plumbing/format/packfile/delta_selector.go]: You should be able to just do `results <- err`, or `results <- dw.fixAndBreakChainsOne(objectsToPack, j) `

**Review by Nivl — APPROVED** [review]
Looks good to me

**Pull request merged**